CONTRIBUTING.md

```diff
@@ -114,6 +114,35 @@ git checkout <your-branch>
 git rebase sfcli/main
 ```
 
+## Displaying intermediate progress to stdout
+If you want to display some additional information to stdout during command execution you should use methods provided by `cli_console`.
+
+`cli_console` provides three methods:
+- `phase` for displaying unindented messages
+- `step` for displaying 2 spaces indented messages
+- `error` for displaying unindented messages
+
+All messaged are displayed with theirs own predefined styles. Styling is provided by rich.
+
+```python
+from snowflake.cli.api.console import cli_console
+
+cli_console.phase("foo")
+cli_console.step("bar")
+cli_console.step("bar")
+cli_console.phase("foo")
+cli_console.phase("foo")
+cli_console.step("bar")
+cli_console.error("Ops")
```

Comment:
How does it differe from raising en exception?

Reply:
+1 I think the most useful feature here would be some kind of warning API instead of error (assuming we all consider error to be something you don't recover from). There are a few use cases for NADE however that would justify issuing a warning to the user, I have some listed on the original JIRA. For errors, exceptions make sense to me.

Reply:
Renamed `error` to `warning`